File: Python/PrimeNumber_1978.py
def make_prime():
    is_prime = [True] * 1001  # 2부터 1000까지의 소수 여부를 저장할 리스트
    is_prime[0] = is_prime[1] = False
    for i in range(2, 1001):
        if is_prime[i]:
            cur = i * 2
            while cur < 1001:
                is_prime[cur] = False
                cur += i
    return is_prime


def main():
    is_prime = make_prime()
    
    # 입력 처리
    import sys
    N = sys.stdin.readline()
    numbers = list(map(int, input().split()))
    
    # 소수 개수 카운트
    ans = sum(1 for num in numbers if is_prime[num])
    print(ans)


if __name__ == "__main__":
    main()

File: Java/PrimeNumber_1978.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class PrimeNumber_1978 {
    private static int N;
    private static boolean[] isPrime;

    public static void main(String[] args) throws Exception {
        makePrime();
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        N = Integer.parseInt(br.readLine());

        StringTokenizer st = new StringTokenizer(br.readLine()," ");
        
        int ans = 0;
        for(int i=0; i<N; i++) {
            int num = Integer.parseInt(st.nextToken());
            if(isPrime[num]) ans++;
        }
        System.out.println(ans);
    }

    public static void makePrime() {
        isPrime = new boolean[1001];
        for(int i=2; i<1001; i++) isPrime[i] = true;
        for(int i=2; i<1001; i++) {
            if(isPrime[i]) {
                int cur = i << 1;
                while(cur < 1001) {
                    isPrime[cur] = false;
                    cur += i;
                }
            }
        }
    }
}
